UnicornRender/utility/sources/Math.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/utility/Math.hpp>
+
+#define GLM_ENABLE_EXPERIMENTAL
+#include <glm/gtx/norm.hpp>
+#include <glm/gtx/quaternion.hpp>
+#include <glm/gtc/quaternion.hpp>
+
+namespace unicorn
+{
+namespace utility
+{
+namespace math
+{
+
+glm::quat RotationBetweenVectors(glm::vec3 start, glm::vec3 dest, glm::vec3 worldX, glm::vec3 worldZ)
+{
+    start = normalize(start);
+    dest = normalize(dest);
+
+    float cosTheta = dot(start, dest);
+    glm::vec3 rotationAxis;
+
+    if (cosTheta < -1 + 0.001f)
+    {
+        rotationAxis = cross(worldZ, start);
+        if (length2(rotationAxis) < 0.01 )
+        {
+            rotationAxis = cross(worldX, start);
+        }
+        rotationAxis = normalize(rotationAxis);
+
+        return angleAxis(glm::radians(180.0f), rotationAxis);
+    }
+
+    rotationAxis = cross(start, dest);
+
+    float s = sqrt((1 + cosTheta) * 2);
+    float invs = 1 / s;
+
+    return glm::quat(
```

Comment:
I suggest you either don't use glm scope resolution or use it everywhere. 
In my opinion, second is better.